src/backend/base/langflow/components/tools/python_repl.py

```diff
@@ -1,96 +1,119 @@
+import ast
 import importlib
 
-from langchain.tools import StructuredTool
-from langchain_core.tools import ToolException
 from langchain_experimental.utilities import PythonREPL
-from loguru import logger
-from pydantic import BaseModel, Field
 
-from langflow.base.langchain_utilities.model import LCToolComponent
-from langflow.field_typing import Tool
-from langflow.inputs import StrInput
+from langflow.custom import Component
+from langflow.io import CodeInput, Output
 from langflow.schema import Data
 
 
-class PythonREPLToolComponent(LCToolComponent):
+class PythonREPLToolComponent(Component):
     display_name = "Python REPL"
-    description = "A tool for running Python code in a REPL environment."
-    name = "PythonREPLTool"
+    description = "A tool for running Python code in a REPL environment with dynamic imports."
     icon = "Python"
+    name = "PythonREPL"
 
     inputs = [
-        StrInput(
-            name="name",
-            display_name="Tool Name",
-            info="The name of the tool.",
-            value="python_repl",
-        ),
-        StrInput(
-            name="description",
-            display_name="Tool Description",
-            info="A description of the tool.",
-            value="A Python shell. Use this to execute python commands. "
-            "Input should be a valid python command. "
-            "If you want to see the output of a value, you should print it out with `print(...)`.",
-        ),
-        StrInput(
-            name="global_imports",
-            display_name="Global Imports",
-            info="A comma-separated list of modules to import globally, e.g. 'math,numpy'.",
-            value="math",
-        ),
```

Comment:
We had imports for security reasons. We need a way to limit what the LLM can do.